plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/PomXmlScanner.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.validation;
+
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.ext.Locator2;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Adds <groupId> element to {@link BannedElement} queue if the Maven plugin is not up to date.
+ */
+class PomXmlScanner extends AbstractScanner {
+
+  private boolean foundBuild;
+  private boolean foundGroupId;
```

Comment:
foundGroupId, foundArtifactId, etc should probably be renamed insideGroupId, insideArtifactId, etc.